spec/ruby/core/exception/full_message_spec.rb

```diff
@@ -46,6 +46,33 @@
     full_message[0].should.end_with?("': Some runtime error (RuntimeError)\n")
   end
 
+  describe "includes details about whether an exception was handled" do
+    describe "RuntimeError" do
+      it "should report as unhandled if message is empty" do
+        err = RuntimeError.new("")
+
+        err.full_message.should =~ /unhandled exception/
+        err.full_message(highlight: true).should =~ /unhandled exception/
+        err.full_message(highlight: false).should =~ /unhandled exception/
```

Comment:
Hmm... I see the point about the default `highlight:` value changing based on whether running with a TTY. But, I also think it looks strange to not test that case since. Especially since that's the most common way to get the message. Oh well.

Reply:
One problem is we report a different source location. Maybe that's a bug in `full_message` that needs to be resolved, but it's a step beyond ensuring we supply a default message. I'll see how involved that might be.

Reply:
I think this might be solved by rebasing on https://github.com/oracle/truffleruby/pull/3263 and moving these specs to `detailed_message_spec.rb`, that should involve no backtrace or source location.

It seems we have some incompatibility in TruffleRuby that if an exception has not been raise-d, full_message includes a backtrace while on CRuby it does not.